src/D2L.CodeStyle.Analyzers/Language/RethrowAnalyzer.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Operations;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace D2L.CodeStyle.Analyzers.Language {
+
+	[DiagnosticAnalyzer( LanguageNames.CSharp )]
+	internal sealed partial class RethrowAnalyzer : DiagnosticAnalyzer {
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
+			=> ImmutableArray.Create( Diagnostics.ShouldRethrow );
+
+		public override void Initialize( AnalysisContext context ) {
+			context.EnableConcurrentExecution();
+			context.RegisterCompilationStartAction( RegisterAnalyzer );
+		}
+
+		private static void RegisterAnalyzer(
+			CompilationStartAnalysisContext context
+		) {
+			context.RegisterOperationAction(
+				Analyze,
+				OperationKind.CatchClause
+			);
+		}
+
+		private static void Analyze(
+			OperationAnalysisContext context
+		) {
+			ICatchClauseOperation operation = context.Operation as ICatchClauseOperation;
+
+			if( operation.ExceptionDeclarationOrExpression == null ) {
+				// try {} catch {}
+				// try {} catch( Exception ) {}

```

Comment:
This pattern could be bad too - is it cool that we're ignoring it? Future thing to consider or what?

Reply:
I see... we are explicitly looking for `throw`s here, got it